app/scripts/modules/core/src/pipeline/pipeline.states.ts

```diff
@@ -92,5 +94,39 @@ module(PIPELINE_STATES, [APPLICATION_STATE_PROVIDER]).config([
     };
 
     applicationStateProvider.addChildState(pipelines);
+
+    const executionsLookup: INestedState = {
+      name: 'executionLookup',
+      url: '/executions/:executionId',
+      params: {
+        executionId: { dynamic: true },
+      },
+      redirectTo: transition => {
+        const { executionId } = transition.params();
+        const executionService: ExecutionService = transition.injector().get('executionService');
+
+        if (!executionId) {
+          return undefined;
+        }
+
+        return Promise.resolve()
```

Comment:
can this `Promise.resolve()` be removed?

```js
        return executionService.getExecution(executionId)
          .then(execution =>
```

Reply:
`redirectTo` expects a `Promise` so needed to convert the `IPromise` from `executionService`.

Is this the right way to do that or is there a better way?